libdeno/BUILD.gn

```diff
@@ -0,0 +1,83 @@
+# Copyright 2018 the Deno authors. All rights reserved. MIT license.
+import("./deno.gni")
+import("//third_party/v8/gni/v8.gni")
+
+# Only functionality needed for libdeno_test and snapshot_creator
+# In particular no flatbuffers, no assets, no rust, no msg handlers.
+# Because snapshots are slow, it's important that snapshot_creator's
+# dependencies are minimal.
+static_library("libdeno") {
+  sources = [
+    "api.cc",
+    "binding.cc",
+    "deno.h",
+    "file_util.cc",
+    "file_util.h",
+    "internal.h",
+  ]
+  public_deps = [
+    "//third_party/v8:v8_monolith",
+  ]
+  configs += [ ":deno_config" ]
+}
+
+executable("snapshot_creator") {
+  sources = [
+    "snapshot_creator.cc",
+  ]
+  deps = [
+    ":libdeno",
+  ]
+  configs += [ ":deno_config" ]
+}
+
+v8_executable("test_cc") {
+  testonly = true
+  sources = [
+    "file_util_test.cc",
+    "libdeno_test.cc",
+    "test.cc",
+  ]
+  deps = [
+    ":create_snapshot_libdeno_test",
+    ":libdeno",
+    "//testing/gtest:gtest",
+  ]
+  data = [
+    "$target_gen_dir/snapshot_libdeno_test.bin",
+  ]
+  snapshot_path = rebase_path(data[0], root_build_dir)
+  defines = [ "SNAPSHOT_PATH=\"$snapshot_path\"" ]
+  configs = [ ":deno_config" ]
+}
+
+# Generates $target_gen_dir/snapshot_libdeno_test.bin
+create_snapshot("libdeno_test") {
+  testonly = true
+  js = "libdeno_test.js"
+}
+
+config("deno_config") {
+  include_dirs = [ "//third_party/v8" ]  # This allows us to v8/src/base/ libraries.
+  configs = [ "//third_party/v8:external_config" ]
+  if (is_debug) {
+    defines = [ "DEBUG" ]
+  }
+
+  # Targets built with the `rust_executable()` template automatically pick up
+  # these dependencies, but those built with `executable()` need them when they
+  # have Rust inputs. Currently, there's only one such target, `test_cc`.
+  if (is_mac) {
+    libs = [ "resolv" ]
+  }
+  if (is_win) {
+    libs = [ "userenv.lib" ]
+  }
+
+  if (is_clang) {
+    cflags = [
+      "-fcolor-diagnostics",
```

Comment:
Ditto. Arguably this would be most appropriate in BUILDCONFIG.gn or compiler.gni, but since we we can't modify that, leave it in //BUILD.gn.

Reply:
Removed completely since we no longer have any executable() targets that depend on rust code.

Reply:
The -fansi-escape-codes thing is unrelated to Rust -- it's used to improve readability of clang output. We have similar flags for rust and tsc invocations that I didn't add so don't tell me these ergonomics don't matter ;)

Reply:
I've added it back.

Reply:
BTW It seems this flag is set in the toolchain 
https://cs.chromium.org/chromium/src/build/config/win/BUILD.gn?l=110&rcl=5814800fa70a7a7a7975c9943cdf0c310873321c

Why do we add it here?

Reply:
> BTW It seems this flag is set in the toolchain
> https://cs.chromium.org/chromium/src/build/config/win/BUILD.gn?l=110&rcl=5814800fa70a7a7a7975c9943cdf0c310873321c
> 
> Why do we add it here?

Because it only adds the flag if the `ANSICON` environment variable is set. It certainly isn't set by default or by anything that I know of.